identity/identity.go

```diff
@@ -30,11 +28,50 @@ func NewIdentityClient(address string, s signer.DynamicSigner) (IdentityClient,
 	return &identityClient{auth: auth, s: s, cm: crypto.NewCryptoManager()}, nil
 }
 
-func (c *identityClient) Register(params params.RegisterIdentityParams) error {
-	return errors.New("not implemented")
+func (c *identityClient) Register(p *params.RegisterIdentityParams) error {
+	did := c.s.GetDid()
+	if p.Challenge != nil {
+		nonce := p.Challenge.Nonce
+		signed, err := c.s.Sign(signer.Authentication, []byte(did+"."+nonce))
+		if err != nil {
+			return err
+		}
+
+		err = c.auth.AuthnWithDidRegister(did, nonce, signed)
+		if err != nil {
+			return err
+		}
+	} else {
+		challenge, stream, err := c.auth.CreateDidRegistrationChallenge(p.Token)
+		if err != nil {
+			return err
+		}
+		nonce := challenge.Nonce
+		signed, err := c.s.Sign(signer.Authentication, []byte(did+"."+nonce))
+		if err != nil {
+			return err
+		}
+
+		err = c.auth.AuthnWithDidRegister(did, nonce, signed)
+		if err != nil {
+			return err
```

Comment:
Would be great to wrap these errors with some messages to be able to track down what went wrong. 
Moreover - are they retryable?

Reply:
@sKudryashov I think the errors should be defined by the API and then the SDK should only do minimal changes (if any) to the error, what would you suggest?

Reply:
I think what I'd be okay with doing is to, in a future issue, create a new error object that has both a code and a message (similarly to the gRPC error code).

What do you think?

Reply:
I think it can be solved by error types or error interfaces, not error codes.

Reply:
@LuisOsta I agree the SDK is a wrapper around the core. The improvements to error handling should happen on the API layer.

Reply:
Resolving this now, though lets definitely investigate in the future how we can improve the error handling experience for consumers of the SDK.

Reply:
That'll be handled in https://github.com/knox-networks/knox-go/issues/26